### PR TITLE
KAFKA-2425; Copy latest docs to kafka repo docs/ directory

### PR DESCRIPTION
**Pull request by omkreddy**
This PR copies the latest kafka docs to kafka repo docs directory. Here I have copied 0.8.3/ directory contents from svn website repo to kafka/docs repository.

Some questions: This PR contains generated javadocs also.  Do we need to copy javadocs here?

### CODE REVIEW

**Comment by omkreddy** [comment]
@gwenshap @junrao @guozhangwang pinging for review

**Comment by ijuma** [comment]
How will we update the docs in the kafka-site repository? Is the idea to copy manually from this one? If so, it would be good to document this somewhere (maybe the wiki?) so that it's clear what the plan is.

**Comment by omkreddy** [comment]
Yes, that is the plan. We will maintain only  technical doc pages 
(api.html, design.html, ecosystem.html, introduction.html, ops.html, upgrade.html
configuration.html, documentation.html, implementation.html, migration.html
quickstart.html, uses.html)

During release, we need to manually copy these files to a version directory (08X/) inside kafka-site repo and update documentation.html, downloads.html pages in kafka-site repo.

After this PR merge, I will update contributing.html

**Comment by ijuma** [comment]
We also need to update the "Release Process" wiki page:

https://cwiki.apache.org/confluence/display/KAFKA/Release+Process

**Comment by omkreddy** [comment]
Sure, I will update the "Release Process" wiki page.

**Comment by omkreddy** [comment]
I removed javadocs from kafka/docs folder.  Generated javadocs needs to be copied to site docs during release process.

It will be good, If we commit this PR at the earliest. So that new 0.8.3 docs can be easily updated by kafka contributors.

**Comment by gwenshap** [comment]
Sorry for the delay and thank you for helping with this.

Two comments:

1) The docs should have a license header (with the Apache license as html comment), and the RAT check should include the docs.
2) We need a gradle task for packaging the docs so we can release a doc package with our usual release and  people with connectivity issues can just get their own private copy of the docs with the release.

These can be done as follow up JIRA.

**Comment by omkreddy** [comment]
Just so we are on the same page, we are maintaining only technical doc pages here, not complete web site.  I will do the required changes in this patch.

**Comment by omkreddy** [comment]
@gwenshap I updated the patch with required changes.  Please have a look at it.

File: build.gradle [hunk 1]
```diff
@@ -313,6 +313,12 @@ project(':core') {
     from sourceSets.test.output
   }
 
+  task publishDocs(type: Jar) {
```
Comment (gwenshap) [inline build.gradle]: I'm not sure JAR is the right way to publish user-facing docs. Most projects either generate tarball, throw the docs into target/site (build/site in our case) or both.

Comment (omkreddy) [inline build.gradle]: Do you mean to say that, docs tar file should be part of kafka distribution and inside kafka/site folder?

Current patch will upload kafka-docs jar file to maven and kafka distribution.

Comment (ijuma) [inline build.gradle]: Gwen, you may be right about the tarball and site, but javadoc artifacts are a standard Maven thing (along with jars and sources jars), eg see here:

http://repo1.maven.org/maven2/org/scala-lang/scala-library/2.11.7/

IDEs and other tools can use those jars.

@omkreddy, will the docs classifier in Gradle generate a jar with the javadoc suffix in the name?

Comment (omkreddy) [inline build.gradle]: @ijuma  We are not discussing about javadocs.  We want  kafka/site documentation pages to be part of release of artifacts.

Comment (ijuma) [inline build.gradle]: Oh, then I agree with Gwen, it's odd to upload that to Maven as a jar. Sorry for the noise.

Comment (omkreddy) [inline build.gradle]: uploaded a new patch which copies docs tarball to kafka distribution.

**Comment by gwenshap** [comment]
LGTM!

**Comment by b-goyal** [comment]
This PR copies the latest kafka docs to kafka repo docs directory. Here I have copied 0.8.3/ directory contents from svn website repo to kafka/docs repository.

Some questions: This PR contains generated javadocs also.  Do we need to copy javadocs here?

**Pull request closed without merging**